From 2759754956be292570723469121f1be917daffbb Mon Sep 17 00:00:00 2001
From: Koushik Dutta <koushd@gmail.com>
Date: Sat, 26 Jun 2010 22:53:55 -0700
Subject: [PATCH 3/3] core: Add support for overriding build.prop settings

Author: Koushik Dutta <koushd@gmail.com>
Date:   Sat Jun 26 22:53:55 2010 -0700

    core: Add support for overriding build.prop settings

    Change-Id: Ie4224d608b7bcb0c2087bd77bc954f8f5f9b6495

Author: LuK1337 <priv.luk@gmail.com>
Date:   Fri Sep 13 03:44:13 2019 +0200

    Make PRODUCT_BUILD_PROP_OVERRIDES bit more powerful

    * This lets devices override stuff like PRODUCT_DEVICE
      for every kind of prop.

    Change-Id: I2afdf0b466bb5a1286b65526bcea98077d1650c0

Change-Id: I665a0281e4f101ec9ac3c5805dccab77d98c0d0b
---
 core/sysprop.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/sysprop.mk b/core/sysprop.mk
index 58e32013e0..f54524d54f 100644
--- a/core/sysprop.mk
+++ b/core/sysprop.mk
@@ -106,7 +106,8 @@ $(2): $(POST_PROCESS_PROPS) $(INTERNAL_BUILD_ID_MAKEFILE) $(API_FINGERPRINT) $(3
 	$(hide) mkdir -p $$(dir $$@)
 	$(hide) rm -f $$@ && touch $$@
 ifneq ($(strip $(7)), true)
-	$(hide) $$(call generate-common-build-props,$(call to-lower,$(strip $(1))),$$@)
+	$(hide) $(PRODUCT_BUILD_PROP_OVERRIDES) \
+	        $$(call generate-common-build-props,$(call to-lower,$(strip $(1))),$$@)
 endif
 	$(hide) $(foreach file,$(strip $(3)),\
 	    if [ -f "$(file)" ]; then\
-- 
2.37.1

